src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -308,7 +318,7 @@ public Heap()
         /// <param name="values">Array elements whose indices are integer</param>
         /// <param name="dict">Array elements whose indices are not integer</param>
         /// <returns></returns>
-        public StackValue AllocateArray(IEnumerable<StackValue> values)
+        public StackValue AllocateArray(StackValue[] values)
```

Comment:
Because internally we will allocate memory based on the size of `values`, and copy items to the destination. If `values` is `IEnumerable`, potentially we have to iterate it twice.